File: go/0349E-两个数组的交集.go
/*
 * @Author: SukiEva
 * @Date: 2021-10-16 15:33:44
 * @Description: https://github.com/SukiEva
 * わたし、気になります！
 */

package main

func intersection(nums1 []int, nums2 []int) []int {
	cnt := make(map[int]bool)
	var ans []int
	for _, v := range nums1 {
		cnt[v] = true
	}
	for _, v := range nums2 {
		if cnt[v] {
			ans = append(ans, v)
			cnt[v] = false
		}
	}
	return ans
}

File: cpp/0349E-两个数组的交集.cpp
/*
 * @Author: SukiEva
 * @Date: 2021-12-14 08:48:59
 * @Description: https://github.com/SukiEva
 * わたし、気になります！
 */
#include <bits/stdc++.h>

#include "include/ListNode.h"
#include "include/TreeNode.h"

using namespace std;

class Solution {
   public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        unordered_map<int, int> mp;
        unordered_set<int> ans;
        for (auto num : nums1) mp[num]++;
        for (auto num : nums2) {
            if (mp[num]) ans.insert(num);
        }
        return vector<int>(ans.begin(), ans.end());
    }
};